Test regression on PyPy3.10: `FAILED tests/test_main.py::test_predicate_repr[pred2-Predicate(math.isfinite)] - AssertionError: assert 'Predicate(isfinite)' == 'Predicate(math.isfinite)'`

The 0.7.0 release introduces a test regression on PyPy3.10:

```pytb
$ pypy3.10 -m pytest
========================================================= test session starts =========================================================
platform linux -- Python 3.10.14[pypy-7.3.16-final], pytest-8.2.0, pluggy-1.5.0
rootdir: /tmp/annotated-types
configfile: pyproject.toml
testpaths: tests
plugins: anyio-4.3.0, subtests-0.12.1, asyncio-0.23.6, xprocess-1.0.1, xdist-3.6.1
asyncio: mode=strict
collected 256 items                                                                                                                   

tests/test_grouped_metadata.py ..                                                                                               [  0%]
tests/test_main.py ............................................................................................................ [ 42%]
............................................................................................................................... [ 92%]
................F..                                                                                                             [100%]

============================================================== FAILURES ===============================================================
_________________________________________ test_predicate_repr[pred2-Predicate(math.isfinite)] _________________________________________

pred = Predicate(isfinite), repr_ = 'Predicate(math.isfinite)'

    @pytest.mark.parametrize(
        "pred, repr_",
        [
            (annotated_types.Predicate(func=a_predicate_fn), "Predicate(a_predicate_fn)"),
            (annotated_types.Predicate(func=str.isascii), "Predicate(str.isascii)"),
            (annotated_types.Predicate(func=math.isfinite), "Predicate(math.isfinite)"),
            (annotated_types.Predicate(func=bool), "Predicate(bool)"),
            (annotated_types.Predicate(func := lambda _: True), f"Predicate({func!r})"),
        ],
    )
    def test_predicate_repr(pred: annotated_types.Predicate, repr_: str) -> None:
>       assert repr(pred) == repr_
E       AssertionError: assert 'Predicate(isfinite)' == 'Predicate(math.isfinite)'
E         
E         - Predicate(math.isfinite)
E         ?           -----
E         + Predicate(isfinite)

tests/test_main.py:162: AssertionError
======================================================= short test summary info =======================================================
FAILED tests/test_main.py::test_predicate_repr[pred2-Predicate(math.isfinite)] - AssertionError: assert 'Predicate(isfinite)' == 'Predicate(math.isfinite)'
==================================================== 1 failed, 255 passed in 1.74s ====================================================
```

I guess PyPy implements it as a builtin without a loadable module.

```
$ python3.10 -c 'import math; print((math.isfinite, math.isfinite.__self__))'
(<built-in function isfinite>, <module 'math' from '/usr/lib/python3.10/lib-dynload/math.cpython-310-x86_64-linux-gnu.so'>)
$ pypy3.10 -c 'import math; print((math.isfinite, math.isfinite.__self__))'
(<built-in function isfinite>, None)
```

CC @mattip, @cfbolz

--- a/.github/workflows/ci.yml
+++ b/.github/workflows/ci.yml
@@ -15,7 +15,7 @@ jobs:
     strategy:
       fail-fast: false
       matrix:
-        python-version: ['3.8', '3.9', '3.10', '3.11', '3.12', '3.13']
+        python-version: ['3.8', '3.9', '3.10', '3.11', 'pypy3.11', '3.12', '3.13']
 
     runs-on: ubuntu-latest
 

--- a/tests/test_main.py
+++ b/tests/test_main.py
@@ -151,12 +151,12 @@ def a_predicate_fn(x: object) -> bool:
 @pytest.mark.parametrize(
     "pred, repr_",
     [
-        (annotated_types.Predicate(func=a_predicate_fn), "Predicate(a_predicate_fn)"),
-        (annotated_types.Predicate(func=str.isascii), "Predicate(str.isascii)"),
-        (annotated_types.Predicate(func=math.isfinite), "Predicate(math.isfinite)"),
-        (annotated_types.Predicate(func=bool), "Predicate(bool)"),
-        (annotated_types.Predicate(func := lambda _: True), f"Predicate({func!r})"),
+        (annotated_types.Predicate(func=a_predicate_fn), ["Predicate(a_predicate_fn)"]),
+        (annotated_types.Predicate(func=str.isascii), ["Predicate(str.isascii)"]),
+        (annotated_types.Predicate(func=math.isfinite), ["Predicate(math.isfinite)", "Predicate(isfinite)"]),
+        (annotated_types.Predicate(func=bool), ["Predicate(bool)"]),
+        (annotated_types.Predicate(func := lambda _: True), [f"Predicate({func!r})"]),
     ],
 )
 def test_predicate_repr(pred: annotated_types.Predicate, repr_: str) -> None:
-    assert repr(pred) == repr_
+    assert repr(pred) in repr_

